internal/restorer/doc.go

```diff
@@ -5,29 +5,20 @@
 // request and avoiding repeated downloads of the same pack. In addition,
 // several pack files are fetched concurrently.
 //
-// Here is high-level pseudo-code of the how the Restorer attempts to achieve
+// Here is high-level pseudo-code of how the Restorer attempts to achieve
 // these goals:
 //
 //   while there are packs to process
 //     choose a pack to process                      [1]
-//     get the pack from the backend or cache        [2]
+//     retrieve the pack from the backend            [2]
 //     write pack blobs to the files that need them  [3]
-//     if not all pack blobs were used
-//       cache the pack for future use               [4]
 //
-// Pack download and processing (steps [2] - [4]) runs on multiple concurrent
-// Goroutines. The Restorer runs all steps [2]-[4] sequentially on the same
-// Goroutine.
+// Retrieval of repository packs (step [2]) and writing target files (step [3])
+// are performed concurrently on multiple goroutines.
 //
-// Before a pack is downloaded (step [2]), the required space is "reserved" in
-// the pack cache. Actual download uses single backend request to get all
-// required pack blobs. This may download blobs that are not needed, but we
-// assume it'll still be faster than getting individual blobs.
-//
-// Target files are written (step [3]) in the "right" order, first file blob
-// first, then second, then third and so on. Blob write order implies that some
-// pack blobs may not be immediately used, i.e. they are "out of order" for
-// their respective target files. Packs with unused blobs are cached (step
-// [4]). The cache has capacity limit and may purge packs before they are fully
-// used, in which case the purged packs will need to be re-downloaded.
+// Implementation does not guarantee order in which blobs are written to the
+// target files and, for example, the last blob of a file can be written to the
+// file before any of the preceeding file blobs. It is therefore possible to
+// have gaps in the data written to the target files if restore fails or
```

Comment:
Not all Filesystems allow multiple open files at different location, would it be possible to accomodate both strategies?

Reply:
Not sure I understand the concern. Are you saying some filesystem only support sequential writes to a file? Specific example that demonstrates the problem will be useful here.

Reply:
you can see the different modes in rclone(which comes to my mind), https://rclone.org/commands/rclone_mount/#file-caching

only the `vfs-cache-mode full` will work with this commit, but as of now it works with `vfs-cache-mode minimal` also

Reply:
I am afraid 0.9.5 is not compatible with `vfs-cache-mode minimal` already. It may decide to close a file, then reopen it with `O_APPEND` some time later. To support `vfs-cache-mode minimal` we need to go back to pre restic 0.9 impl. I suggest we treat that as a separate feature request.

Reply:
@ifedorenko Can you possibly reference the commit/PR/change/whatever in 0.9.5 that introduced the incompatibility with `vfs-cache-mode minimal`? Thanks!

Reply:
PR #1719 and commit 1869930d95d8c9c1f49e042b5379c0aab72aa353 specifically. 

This is where restore (re)opens files with O_APPEND https://github.com/restic/restic/blob/af200157258579d286d335edef05f0279741602d/internal/restorer/fileswriter.go#L56

Reply:
We'll leave this for now due to the reasons mentioned by @ifedorenko. Let it be a separate feature request if needed.